src/lib/tools/session/index.ts

```diff
@@ -0,0 +1,97 @@
+import { JWTPayload, jwtVerify, SignJWT } from "jose";
+import { cookies } from "next/headers";
+
+export type SessionPayload = {
+  userId: string;
+};
+
+export type JWTSessionPayload = SessionPayload & JWTPayload;
+
+// Check for the environment variable for the secret key, throw error if not set
+if (!process.env.SESSION_SECRET) {
+  throw new Error(
+    "SESSION_SECRET is not set. Please set the SESSION_SECRET environment variable.",
+  );
+}
+
+// Environment variable for the secret key
+export const SESSION_SECRET = process.env.SESSION_SECRET;
+export const SESSION_COOKIE_NAME = "session";
+export const ERROR_INVALID_SESSION_SCHEMA =
+  "Deserialized session cookie has an invalid schema";
+
+export async function createSession(payload: SessionPayload): Promise<string> {
+  const encoder = new TextEncoder();
+  const jwt = await new SignJWT({ userId: payload.userId })
+    .setProtectedHeader({ alg: "HS256", typ: "JWT" })
+    .setIssuedAt()
+    .setExpirationTime("30d")
+    .sign(encoder.encode(SESSION_SECRET));
+
+  return jwt;
+}
+
+/**
+ * Verifies the signed token string and
+ * deserializes it into a {@link JWTPayload} object
+ * with properties from {@link SessionPayload}.
+ *
+ * @returns null if the token cannot be deserialized.
+ * @throws {Error} If the token cannot be deserialized.
+ */
+export async function verifySession(
+  token: string,
+): Promise<JWTSessionPayload | null> {
+  const encoder = new TextEncoder();
+
+  try {
+    const { payload } = await jwtVerify(token, encoder.encode(SESSION_SECRET), {
+      algorithms: ["HS256"],
+    });
+
+    if (!("userId" in payload)) {
+      throw new Error(ERROR_INVALID_SESSION_SCHEMA);
+    }
+
+    return payload as JWTSessionPayload;
+  } catch (error) {
+    if (error instanceof Error) {
+      if (error.name === "JWSSignatureVerificationFailed") {
+        throw new Error(ERROR_INVALID_SESSION_SCHEMA);
```

Comment:
So we are catching all errors except for this one?

Reply:
Should we just catch everything and always return null on error?

Reply:
Well, we are logging the error already, so I thought it wasn't necessary to rethrow it.

Reply:
Right, I wonder if we should re-throw this one either?

Reply:
Well, that error is thrown when a cookie is malformed.